### ISSUE TITLE
SVG's not displaying consistently in elementor editor

### ISSUE THREAD

**Issue by mdrideout** [elementor/elementor#9658]
- [ x ] I have searched for similar issues in both open and closed tickets and cannot find a duplicate.
- [ x ] The issue still exists against the latest stable version of Elementor.

## Description

When uploading an SVG to the "icon box" block, the SVG won't appear while editing that block, and it won't appear until you click out to a different block. Sometimes, you have to hit "update", and then reload to get it to appear.

I am downloading my icons as SVG from flaticon. 

## Steps to reproduce

1. Add an icon box to your page
2. Use the "upload svg" option when adding an icon
3. Select or upload an SVG and hit insert media
4. The SVG won't appear until you click to another block, or sometimes, until after you reload the editor.

## Isolating the problem

- [ x ] This bug happens with only Elementor plugin active (and Elementor Pro).
- [ x ] This bug happens with a default WordPress theme active.
- [ x ] I can reproduce this bug consistently using the steps above.

## Environment

<details>
<summary>System Info</summary>
```
					
== Server Environment ==
	Operating System: Linux
	Software: Apache
	MySQL version: Percona Server (GPL), Release '30', Revision '8916819' v5.7.27-30
	PHP Version: 7.3.11-1+ubuntu18.04.1+deb.sury.org+1
	PHP Max Input Vars: 1000
	PHP Max Post Size: 100M
	GD Installed: Yes
	ZIP Installed: Yes
	Write Permissions: All right
	Elementor Library: Connected

== WordPress Environment ==
	Version: 5.3
	Site URL: https://dirigosafety.wpengine.com
	Home URL: http://dirigosafety.wpengine.com
	WP Multisite: No
	Max Upload Size: 50 MB
	Memory limit: 512M
	Permalink Structure: /%postname%/
	Language: en-US
	Timezone: 0
	Debug Mode: Inactive

== Theme ==
	Name: Dirigo Safety
	Version: 1.0.0
	Author: Matthew Rideout
	Child Theme: Yes
	Parent Theme Name: Astra
	Parent Theme Version: 2.1.3
	Parent Theme Author: Brainstorm Force

== User ==
	Role: administrator
	WP Profile lang: en_US
	User Agent: Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_1) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/78.0.3904.97 Safari/537.36

== Active Plugins ==
	Astra Pro
		Version: 2.1.4
		Author: Brainstorm Force

	Customizer Search
		Version: 1.1.3
		Author: Brainstorm Force

	Duplicate Post
		Version: 3.2.3
		Author: Enrico Battocchi

	Elementor
		Version: 2.7.5
		Author: Elementor.com

	Elementor Pro
		Version: 2.7.3
		Author: Elementor.com

	LearnDash LMS
		Version: 3.1
		Author: LearnDash

	LearnDash ProPanel
		Version: 2.1.3
		Author: LearnDash

	Premium Addons for Elementor
		Version: 3.8.9
		Author: Leap13

	Premium Addons PRO
		Version: 1.7.8
		Author: Leap13

	Safe SVG
		Version: 1.9.6
		Author: Daryll Doyle

	Ultimate Addons for Elementor
		Version: 1.20.0
		Author: Brainstorm Force

	Ultimate Addons for Gutenberg
		Version: 1.14.1
		Author: Brainstorm Force


== Must-Use Plugins ==
	Force Strong Passwords - WPE Edition
		Version: 1.6.4
		Author: Jason Cosper

	WP Engine Security Auditor
		Version: 1.0.8
		Author: wpengine

	WP Engine System
		Version: 3.2.2
		Author: WP Engine

	WPE Sign On Plugin
		Version: 1.1.0
		Author: WP Engine


== Log ==
	
JS: showing 2 of 2JS: 2019-11-12 20:22:01 [error X 2][https://dirigosafety.wpengine.com/wp-content/plugins/elementor/assets/js/editor.min.js?ver=2.7.5:2:62481] Cannot read property \'hoverIntent\' of undefined 
JS: 2019-11-13 17:18:41 [error X 3][https://dirigosafety.wpengine.com/wp-content/plugins/elementor/assets/js/editor.min.js?ver=2.7.5:7:20270] Cannot read property \'noUiSlider\' of undefined 

PHP: showing 3 of 3PHP: 2019-11-12 22:40:55 [warning X 1][/nas/content/live/dirigosafety/wp-content/plugins/elementor-pro/modules/assets-manager/asset-types/icons/custom-icons.php::180] unlink(003-staff.svg): No such file or directory [array (
  'trace' => '
#0: Elementor\Core\Logger\Manager -> shutdown()
',
)]
PHP: 2019-11-12 22:46:06 [warning X 1][/nas/content/live/dirigosafety/wp-content/plugins/elementor-pro/modules/assets-manager/asset-types/icons/custom-icons.php::180] unlink(Flaticon.woff2): No such file or directory [array (
  'trace' => '
#0: Elementor\Core\Logger\Manager -> shutdown()
',
)]
PHP: 2019-11-19 16:46:21 [notice X 2][/nas/content/live/dirigosafety/wp-content/plugins/elementor/includes/elements/column.php::1002] Undefined index: _column_size [array (
  'trace' => '
#0: Elementor\Core\Logger\Manager -> shutdown()
',
)]



== Elements Usage ==
	
	wp-page
		button : -31
		divider : -8
		form : 1
		heading : -51
		icon-box : -11
		image : -20
		spacer : -14
		testimonial-carousel : 1
		text-editor : -17

				
```
</details>

**Comment by Noob86** [comment]
When using SVGs on the website, they frequently don't load on first page load. When reloading/refreshing the page they come up. These SVGs are used as images and uploaded using the icon-widget.

Some SVGs are used with links, e.g. the website logo. The link area is still clickable, but the SVG is not shown. Is there any way to force the SVG to load?

**Comment by lior-sh** [comment]
Hi @mdrideout,

Thanks for reporting, we couldn't reproduce the issue. Can you please add more info like a video that shows the issue?

**Comment by lakalma** [comment]
I've got same problem (suddenly, after using quite more SVG without any problem). It seems to work telling it a width,

**Comment by ATaylor66** [comment]
Hi, was this ever solved, please? I'm having the same issue with a simple SVG I made myself in Affinity Designer. I'm using Elementor Pro, both Elementor and WP are fully updated.

To add more details I'm using the svg in a footer in an image widget set to 300x300px. It seems to work fine but as soon as I click on any other widget to edit them it disappears. I also doesn't show up in the front end.

**Comment by Yaizsk0** [comment]
From what I've seen, you need to select complete image size in order for SVG to display correctly, otherwise it will show up with 1px width and height.

**Comment by voidshaman** [comment]
> From what I've seen, you need to select complete image size in order for SVG to display correctly, otherwise it will show up with 1px width and height.

almost 2 years later and you saved me a headache

**Comment by JEricaM** [comment]
Still have this problem.

**Comment by xavier388** [comment]
I had the same issue and what worked for me was to turn the image into a PNG and then run through different free online PNG-to-SVG converters. Some didn't work but I found success with 'pngtosvg .com'...I have no affiliation with them. I only had a very basic image so not sure if you had a very detailed one, how good it would be. I'm not a technical designer so can't comment on why it worked compared to others from a code perspective and I had run the original SVG (that wasn't appearing) through multiple 'SVG cleaners' but nothing was working. After pulling my hair out over this I finally got my SVG icon to appear on my Elementor site, so just adding this info here so it may help someone else in the future.
